cosmo_tester/test_suites/bootstrap_based_tests/inplace_upgrade_test.py

```diff
@@ -43,17 +44,13 @@ def test_inplace_upgrade(cfy,
     manager = cluster.managers[0]
     snapshot_name = 'inplace_upgrade_snapshot'
     snapshot_path = join(str(module_tmpdir), snapshot_name) + '.zip'
-    hello_world = HelloWorldExample(
-        cfy, manager, attributes, ssh_key, logger, module_tmpdir)
-    hello_world.blueprint_file = 'openstack-blueprint.yaml'
-    hello_world.inputs.update({
-        'agent_user': attributes.centos7_username,
-        'image': attributes.centos7_image_name,
-    })
-    hello_world.upload_blueprint()
-    hello_world.create_deployment()
-    hello_world.install()
-    hello_world.verify_installation()
+    hellos = get_hello_worlds(cfy, manager, attributes, ssh_key,
```

Comment:
Why are we using `get_hello_worlds` here, and not the `hellos` fixture?

Reply:
I was asking myself the same question just now. It's because the inplace upgrade doesn't just have one manager (VM), it has two managers (well, VMs). Rather than hacking up a workaround to make the hello_worlds fixture work with both a single manager and a list of managers, I migrated the code out like this. I'll add a comment to the code to explain that.

Reply:
Cool, sounds fair